modules/fedimint-ln-client/src/lib.rs

```diff
@@ -172,24 +214,31 @@ impl LightningClientExt for Client {
         Ok(instance.api.fetch_gateways().await?)
     }
 
-    async fn pay_bolt11_invoice(
-        &self,
-        invoice: Invoice,
-    ) -> anyhow::Result<(OperationId, ContractId)> {
+    async fn pay_bolt11_invoice(&self, invoice: Invoice) -> anyhow::Result<(PayType, ContractId)> {
         let (lightning, instance) = self.get_first_module::<LightningClientModule>(&KIND);
-        let operation_id = OperationId(invoice.payment_hash().into_inner());
-        let active_gateway = self.select_active_gateway().await?;
+        let payment_hash = invoice.payment_hash();
+        let operation_id = OperationId(payment_hash.into_inner());
 
-        let (output, contract_id) = lightning
-            .create_outgoing_output(
-                operation_id,
-                instance.api,
-                invoice.clone(),
-                active_gateway,
-                self.get_config().await.federation_id,
-                rand::rngs::OsRng,
-            )
-            .await?;
+        let gateways = self.fetch_registered_gateways().await?;
```

Comment:
Note: this adds an extra network round trip to a threshold of guardians every time we pay an invoice.

Reply:
we could cache known gateways. adopting [this proposal](https://github.com/fedimint/fedimint/issues/2618#issuecomment-1587369593) would completely save us the network trip if we never have to look through registered gateways list to identify internal payments